### PR TITLE
custom-elements: Add a test for upgrade failure due to non-null shado…

### PR DESCRIPTION
**Pull request by tkent-google**
…w root and disabledFeatures

Specification PR: https://github.com/whatwg/html/pull/4673

### CODE REVIEW

**Review by hayatoito — APPROVED** [review]
LGTM.

File: custom-elements/upgrading.html [hunk 1]
```diff
@@ -185,6 +185,32 @@
 
 });
 
+test(() => {
+    class InternalsDisabledElement extends HTMLElement {
+      static get disabledFeatures() { return ['internals']; }
+    }
+    let element = document.createElement('internals-disabled');
+    customElements.define('internals-disabled', InternalsDisabledElement);
+    element.attachInternals();
```
Comment (EdgarChen) [inline custom-elements/upgrading.html]: Per step 4 of https://html.spec.whatwg.org/multipage/custom-elements.html#dom-attachinternals, attachInternals() should throw `NotSupportedError`, so it looks like this is an invalid case?

Comment (tkent-google) [reply custom-elements/upgrading.html]: Oh, you're right.  We can't do attachInternals() successfully before upgrading.
Removed the test.

**Review by annevk — APPROVED** [review]
I think It'd be ideal if we also caught the NotSupportedError using window.onerror, though maybe that's too involved with WPT?

**Comment by tkent-google** [comment]
> I think It'd be ideal if we also caught the NotSupportedError using window.onerror, though maybe that's too involved with WPT?

That's a good idea.  Done.

**Review by annevk — APPROVED** [review]

**Pull request merged**